src/Generating/FinishGen.cpp

```diff
@@ -801,6 +805,11 @@ void cFinishGenSprinkleFoliage::GenFinish(cChunkDesc & a_ChunkDesc)
 					)
 					{
 						a_ChunkDesc.SetBlockType(x, ++Top, z, E_BLOCK_CACTUS);
+
+						for (int i = 0; i < (x % m_MaxCactusHeight); i++)
```

Comment:
`(x % m_MaxCactusHeight)` isn't the ideal way to do this because the cacti will line up in sawtooth patterns.  You should use `m_Noise` to make it look more random.